client/src/services/tournament.js

```diff
@@ -49,6 +50,21 @@ class Tournament {
 			})
 	}
 
+	teamsMatches(teamNumber){
+		if(this._teamsMatches && this._teamsMatches[teamNumber]){
+			return Promise.resolve(this._teamsMatches[teamNumber])

```

Comment:
Very good job with the promises!
Something I just learned today, is that it's better to save the promise, and always return it, instead of returning `Promise.resolve`. LIke this:
```javascript
const _cachedPromise
function promiseCache() {
  if(!_cachedPromise) {
    _chacedPromise = ... // caclulating the promise
  }
  return _cachedPromise
}
```

Reply:
This Promise.resolve format is used in all the functions in tournament. Should I just overhaul all of them?